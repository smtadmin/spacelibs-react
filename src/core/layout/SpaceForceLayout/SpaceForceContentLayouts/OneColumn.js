/*
 * File: OneColumn
 * Version: 0.1.0
 * Project: react-test
 * Description: Component for rendering the contents passed to it in a single column
 * File Created: Friday, 29th January 2021 10:15 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Friday, 19th February 2021 11:33 am
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';

/**
 * Component for rendering the contents passed to it in a single column
 */
class OneColumn extends React.Component {

    /**
     * Creates an instance of the OneColumn class
     * @param {*} props 
     */
    constructor(props) {
        super();
    }

    /**
     * Returns a OneColumn component
     */
    render() {
        return (
            <Container className="single-column-container">
                <Row>
                    <Col className="col-12">
                        {this.props.column}
                    </Col>
                </Row>
            </Container>
        )
    }

}

export default OneColumn;