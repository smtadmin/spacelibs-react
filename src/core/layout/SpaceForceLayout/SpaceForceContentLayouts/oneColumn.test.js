/*
 * File: oneColumn.test
 * Version: 0.1.0
 * Project: react-test
 * Description: Testing suite for the SpaceForceContentLayout OneColumn Layout
 * File Created: Monday, 8th February 2021 12:30 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Friday, 12th February 2021 9:14 am
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import OneColumn from './OneColumn';
import React from 'react';
import { render } from '@testing-library/react';

/**
 * It renders the OneColumn element without crashing
 */
it("Renders without crashing", () => {

    const element = render(<OneColumn  />);
    expect(element).toBeTruthy();

});

/**
 * It renders a single element into the OneColumn layout correctly
 */
it("Renders The Layout Correctly With One Element", () => {

    const element = render(<OneColumn colOne={<div>Hello World</div>} />);
    expect(element).toMatchSnapshot();
})

/**
 * It renders multiple elements into the OneColumn layout correctly
 */
it("Renders The Layout Correctly With Multiple Element", () => {

    const element = render(<OneColumn colOne={[<div>One</div>, <div>Two</div>, <div>Three</div>]} />);
    expect(element).toMatchSnapshot();
})