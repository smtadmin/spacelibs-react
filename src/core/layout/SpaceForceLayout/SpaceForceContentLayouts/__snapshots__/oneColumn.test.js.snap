// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders The Layout Correctly With Multiple Element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="single-column-container mb-5 mt-5 container"
      >
        <div
          class="row"
        >
          <div
            class="col-12 mb-5 col"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="single-column-container mb-5 mt-5 container"
    >
      <div
        class="row"
      >
        <div
          class="col-12 mb-5 col"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders The Layout Correctly With One Element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="single-column-container mb-5 mt-5 container"
      >
        <div
          class="row"
        >
          <div
            class="col-12 mb-5 col"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="single-column-container mb-5 mt-5 container"
    >
      <div
        class="row"
      >
        <div
          class="col-12 mb-5 col"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
